src/main/java/xbot/common/controls/actuators/XCANMotorController.java

```diff
@@ -223,33 +235,212 @@ public void periodic() {
+    }
+
+    /**
+     * Override the position of the motor controller.
+     * @param position The new position to set.
+     * @apiNote Angle scaling factors configured on the motor controller are applied.
+     */
     public void setPosition(Angle position) {
+        if (angleScaleFactor == null) {
+            setRawPosition(position);
+        } else {
+            setRawPosition(position.timesConversionFactor(invertRatio(angleScaleFactor)));
+        }
+    }
+
+    /**
+     * Override the position of the motor controller.
+     * @param position The new position to set.
+     * @apiNote Angle scaling factors configured on the motor controller are ignored.
+     */
+    public abstract void setRawPosition(Angle position);
+
+    /**
+     * Set the target position for the motor controller.
+     * @param position The target position to set.
+     * @apiNote Angle scaling factors configured on the motor controller are applied.
+     */
+    public void setPositionTarget(Angle position) {
         setPositionTarget(position, MotorPidMode.DutyCycle);
     }
 
+    /**
+     * Set the target position for the motor controller.
+     * @param position The target position to set.
+     * @param mode The PID mode to use when setting the target position.
+     * @apiNote Angle scaling factors configured on the motor controller are applied.
+     */
     public void setPositionTarget(Angle position, MotorPidMode mode) {
         setPositionTarget(position, mode, 0);
     }
 
-    public abstract void setPositionTarget(Angle position, MotorPidMode mode, int slot);
+    /**
+     * Set the target position for the motor controller.
+     * @param position The target position to set.
+     * @param mode The PID mode to use when setting the target position.
+     * @param slot The PID slot to use when setting the target position.
+     * @apiNote Angle scaling factors configured on the motor controller are applied.
+     */
+    public void setPositionTarget(Angle position, MotorPidMode mode, int slot) {
+        if (angleScaleFactor == null) {
+            setRawPositionTarget(position, mode, slot);
+        } else {
+            setRawPositionTarget(position.timesConversionFactor(invertRatio(angleScaleFactor)), mode, slot);
+        }
+    }
 
+    /**
+     * Set the target position for the motor controller.
+     * @param position The target position to set.
+     * @apiNote Angle scaling factors configured on the motor controller are ignored.
+     */
+    public void setRawPositionTarget(Angle position) {
+        setRawPositionTarget(position, MotorPidMode.DutyCycle);
+    }
+
+    /**
+     * Set the target position for the motor controller.
+     * @param position The target position to set.
+     * @param mode The PID mode to use when setting the target position.
+     * @apiNote Angle scaling factors configured on the motor controller are ignored.
+     */
+    public void setRawPositionTarget(Angle position, MotorPidMode mode) {
```

Comment:
⭐ ⭐ After reading a lot of this code, I keep getting confused about what's a raw position vs scaled position/distance. Especially with everything being Angle. A low hanging fruit improvement could be renaming variables like this `position` to `rawPosition`. Longer term I was wondering if we could use the type system to help differentiate these different Angle's into separate types so you can't use the wrong one?

Reply:
I think I fixed this in all the cases to change the parameter name.